openedx/openedx-platform
# feat: Make courseware URLs more human-readable

## Description

Describe what this pull request changes, and why. Include implications for people using this change.
Design decisions and their rationales should be documented in the repo (docstring / ADR), per
[OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

This change impacts the Learner.

Before:

After:

## Supporting information

Jira issue: [TNL-8511](https://openedx.atlassian.net/browse/TNL-8511)

## Testing instructions

- Navigate to the home page of a course.
- Click on the "Start Course"/"Resume Course" button.
- The page will fully load.
- The URL in the address bar should follow this pattern:
  - host/c/:courseId/:sequenceHash/:unitHash
  Ex: http://localhost:2000/c/course-v1:edx+kris1+2021_T1/YmE3OGQwNzMyYWE2/ZmY0MzRjMTViYjBm
-  Navigate back to the home page.
- Click on a specific section from the outline.
- The page will fully load.
- The URL should match the same formatting as above.
- In the unit view, click on the bookmark button.
- The bookmark saves.
- Navigate back to the home page.
- Click on the bookmark page.
- All bookmarks are seen, even the one just added.
- Navigate back to the previous unit.
- Click on the bookmark button.
- The bookmark is removed.
- Go back to the bookmark page.
- The removed bookmark is no longer listed.

## Deadline

August 11, 2021: Must be completed before the end of the internship

## Other information

This change is dependent on Frontend-App-Learning [PR-540](https://github.com/edx/frontend-app-learning/pull/540)
